components/async-operation-status-mgt/org.wso2.carbon.identity.framework.async.operation.status.mgt/src/main/java/org/wso2/carbon/identity/framework/async/operation/status/mgt/internal/dao/impl/AsyncOperationOperationStatusMgtDAOImpl.java

```diff
@@ -0,0 +1,584 @@
+                            statement.setString(OPERATION_ID, context.getOperationId());
+                            statement.setString(RESIDENT_RESOURCE_ID, context.getOperationInitiatedResourceId());
+                            statement.setString(TARGET_ORG_ID, context.getTargetOrgId());
+                            statement.setString(STATUS, context.getUnitOperationStatus());
+                            statement.setString(STATUS_MESSAGE, context.getStatusMessage());
+                            statement.setTimeStamp(CREATED_AT, currentTimestamp, null);
+                            statement.addBatch();
+                        }
+                    }, null));
+        } catch (TransactionException e) {
+            throw handleServerException(ERROR_WHILE_PERSISTING_ASYNC_OPERATION_STATUS_UNIT, e);
+        }
+    }
+
+    @Override
+    public List<OperationResponseDTO> getOperations(String requestInitiatedOrgId, Integer limit,
+                                                    List<ExpressionNode> expressionNodes) throws
+            AsyncOperationStatusMgtException {
+
+        FilterQueryBuilder filterQueryBuilder = buildFilterQuery(expressionNodes, CREATED_TIME_FILTER);
+        String sqlStmt = getOperationsStatusSqlStmt(filterQueryBuilder);
+
+        List<OperationResponseDTO> operationRecords;
+        NamedJdbcTemplate namedJdbcTemplate = Utils.getNewTemplate();
+        try {
+            operationRecords = namedJdbcTemplate.executeQuery(sqlStmt,
+                (resultSet, rowNumber) -> {
+                    try {
+                        return createOperationResponseDTO(resultSet);
+                    } catch (DataAccessException e) {
+                        throw new RuntimeException(e);
+                    }
+                },
+                namedPreparedStatement -> {
+                    setFilterAttributes(namedPreparedStatement, filterQueryBuilder.getFilterAttributeValue(),
+                            filterQueryBuilder.getTimestampFilterAttributes());
+                    namedPreparedStatement.setInt(LIMIT, limit);
+                    namedPreparedStatement.setString(INITIATED_ORG_ID, requestInitiatedOrgId);
+                });
+        } catch (DataAccessException e) {
+            throw handleServerException(ERROR_WHILE_RETRIEVING_ASYNC_OPERATION_STATUS, e);
+        }
+        return operationRecords;
+    }
+
+    @Override
+    public OperationResponseDTO getOperation(String operationId, String requestInitiatedOrgId) throws
+            AsyncOperationStatusMgtException {
+
+        OperationResponseDTO operationRecord;
+        NamedJdbcTemplate jdbcTemplate = Utils.getNewTemplate();
+        try {
+            operationRecord = jdbcTemplate.fetchSingleRecord(GET_OPERATION, (resultSet, rowNumber) -> {
+                try {
+                    return createOperationResponseDTO(resultSet);
+                } catch (DataAccessException e) {
+                    throw new RuntimeException(e);
```

Comment:
We should not throw a generic RuntimeException() from here. Define a RuntimeException class specific for this component and use it.
Also make sure to catch it below and handle it properly.

Refer: https://github.com/wso2/carbon-identity-framework/blob/825599fead25223ffe2b36376b50785a02cb0856/components/certificate-mgt/org.wso2.carbon.identity.certificate.management/src/main/java/org/wso2/carbon/identity/certificate/management/dao/impl/CertificateManagementDAOImpl.java#L89